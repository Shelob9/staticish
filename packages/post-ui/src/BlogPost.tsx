import React, { Fragment } from 'react';
import Article from './Primative/Article';
import Footer from './Primative/Footer';
import PostTags from './PostTags';
import PostContent, { renderPropForContent } from './PostContent';
import FeaturedImage, { renderFeaturedImage } from './FeaturedImage';
import PostHeader from './PostHeader';
import { PostAuthor, Image } from './types';
import { IconButton } from '@wordpress/components';
import { Post } from '@staticish/wp-api-to-static';
function AuthorProfile(props: { author: PostAuthor }) {
  const { author } = props;
  const { avatar, name, description, link } = author;

  return (
    <div className="flex w-full items-center font-sans px-4 py-12">
      <img
        className="w-10 h-10 rounded-full mr-4"
        src={avatar.src}
        alt={avatar.alt}
      />
      <div className="flex-1 px-2">
        <p className="text-base font-bold text-base md:text-xl leading-none mb-2">
          {name}
        </p>
        {description && (
          <p className="text-gray-600 text-xs md:text-base">{description}</p>
        )}
        {link && (
          <p className="text-gray-600 text-xs md:text-base">
            <a
              className="text-teal-500 no-underline hover:underline"
              href={link.href}
            >
              {link.text}
            </a>
          </p>
        )}
      </div>
      <div className="justify-end">
        <button className="bg-transparent border border-gray-500 hover:border-teal-500 text-xs text-gray-500 hover:text-teal-500 font-bold py-2 px-4 rounded-full">
          Read More
        </button>
      </div>
    </div>
  );
}

function PostTitle(props: { title: string }) {
  return (
    <PostHeader
      level={1}
      tw="font-bold font-sans break-normal text-gray-900 pt-6 pb-2 text-3xl md:text-4xl"
      title={props.title}
    />
  );
}

/**
 * Props for the "renderTitle" render prop.
 */
export type renderTitleProps = {
  title: string;
  post: Post;
};
type renderPropForTitle = (props: renderTitleProps) => JSX.Element;

export function PostTop(props: {
  post: Post;
  renderTitle?: (props: renderTitleProps) => JSX.Element;
  editMode?: boolean;
  onClickSettings?: () => void;
  renderBelowTitle?: () => JSX.Element;
  renderFeatured?: renderFeaturedImage;
}) {
  const { post, onClickSettings, renderFeatured } = props;
  const { title } = post;
  const featured: Image = {
    src: 'http://placekitten.com/1280/640',
    alt: 'A cat',
  };
  return (
    <Fragment>
      <FeaturedImage featured={featured} renderFeatured={renderFeatured} />
      <div
        className="w-full px-4 md:px-6 text-xl text-gray-800 leading-normal"
        style={{ fontFamily: 'Georgia,serif' }}
      >
        <div className="font-sans">
          <div className="text-base md:text-sm text-teal-500 font-bold inline-block">
            <span>
              {props.renderTitle ? (
                props.renderTitle({
                  title: title.rendered,
                  post,
                })
              ) : (
                <PostTitle title={title.rendered} />
              )}
              {props.renderBelowTitle ? (
                React.createElement(props.renderBelowTitle, {
                  post,
                })
              ) : (
                <p className="text-sm md:text-base font-normal text-gray-600">
                  Published {published}
                </p>
              )}
            </span>
          </div>
          {onClickSettings && (
            <div className={'inline-block'}>
              <IconButton
                icon={'admin-generic'}
                label={'Settings'}
                onClick={onClickSettings}
              />
            </div>
          )}
        </div>
      </div>
    </Fragment>
  );
}
export default function(props: {
  post: Post;
  renderTitle?: renderPropForTitle;
  renderContent?: renderPropForContent;
  onClickSettings?: () => void;
  renderBelowTitle?: () => JSX.Element;
  renderFeatured?: renderFeaturedImage;
}) {
  const {
    post,
    renderTitle,
    renderContent,
    onClickSettings,
    renderBelowTitle,
    renderFeatured,
  } = props;
  const { tags, author, content } = post;
  return (
    <Article tw="container w-full md:max-w-3xl mx-auto pt-20">
      <hr className="border-b-2 border-gray-400 mb-8 mx-4" />
      <PostTop
        post={props.post}
        renderTitle={renderTitle}
        onClickSettings={onClickSettings}
        renderBelowTitle={renderBelowTitle}
        renderFeatured={renderFeatured}
      />
      <PostContent content={content.rendered} renderContent={renderContent} />
      <Footer>
        {undefined !== tags && tags.length && (
          <div className="text-base md:text-sm text-gray-500 px-4 py-6">
            <PostTags tags={tags} />
          </div>
        )}
        <AuthorProfile author={author} />
      </Footer>
    </Article>
  );
}
