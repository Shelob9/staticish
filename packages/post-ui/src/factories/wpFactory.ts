import { WpApiPost } from '@staticish/wp-api-to-static';
import { WpPost, WpMedia } from 'src/wpTypes';
import { tag, PostAuthor, Image } from 'src/types';
import {
  WpApiUser,
  WpApiMedia,
  WpApiTaxonomy,
} from '@staticish/wp-api-to-static/dist/wpTypes';

/**
 * Getters for data not included in the WP-API post response
 */
type getters = {
  author: (post: WpApiPost) => WpApiUser;
  featured: (post: WpApiPost) => WpApiMedia;
  tags: (post: WpApiPost) => Array<WpApiTaxonomy>;
  published: (post: WpApiPost) => string;
};

export const wpTag = (fromApi: WpApiTaxonomy): tag => {
  return {
    label: fromApi.name,
    slug: fromApi.slug,
  };
};

export const wpTags = (fromApi: Array<WpApiTaxonomy>): Array<tag> => {
  return fromApi.map(wpTag);
};
const findAvatar = (fromApi: WpApiUser): Image => {
  const avatars = fromApi.avatar_urls;
  const sizes = Object.keys(avatars).map((key: string) => parseInt(key, 10));
  const url = avatars[Math.max(...sizes)];
  return {
    src: url,
    alt: `Avatar for ${fromApi.name}`,
  };
};

/**
 * Format featured image object
 * @param fromApi
 */
export const wpFeatured = (fromApi: WpApiMedia): WpMedia => {
  return {
    src: fromApi.source_url,
    alt: fromApi.alt_text,
    height: fromApi.media_details ? fromApi.media_details.height : undefined,
    width: fromApi.media_details ? fromApi.media_details.width : undefined,
  };
};
/**
 * Prepare author objects for this UI system
 *
 * @param fromApi
 */
export const wpAuthor = (fromApi: WpApiUser): PostAuthor => {
  return {
    name: fromApi.name,
    avatar: findAvatar(fromApi),
    description: fromApi.description,
    link: {
      href: fromApi.link,
      text: name,
    },
  };
};

/**
 * Prepare post objects for this UI system
 *
 * @param fromApi
 * @param getters
 */
const wpPost = (fromApi: WpApiPost, getters: getters): WpPost => {
  const { id, slug, content, title, excerpt } = fromApi;

  return {
    id: id.toString(),
    slug,
    title,
    content,
    excerpt,
    featured: wpFeatured(getters.featured(fromApi)),
    author: wpAuthor(getters.author(fromApi)),
    published: getters.published(fromApi),
    tags: wpTags(getters.tags(fromApi)),
  };
};

export default function(
  getters: getters
): {
  convertPost(fromApi: WpApiPost): WpPost;
} {
  return {
    convertPost(fromApi: WpApiPost): WpPost {
      return wpPost(fromApi, getters);
    },
  };
}
