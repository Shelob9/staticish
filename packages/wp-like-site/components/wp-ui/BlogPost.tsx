import React, { Fragment } from "react";
import Article from "./Primative/Article";
import Footer from "./Primative/Footer";
import PostTags from "./PostTags";
import PostContent, { renderPropForContent } from "./PostContent";
import FeaturedImage, { renderFeaturedImage } from "./FeaturedImage";
import PostHeader from "./PostHeader";
import { PostAuthor, Image } from "./types";
import { WpPost, WpAuthor } from "./wpTypes";
import Author from "./Author";

//Copypasta!
const addAuthorLink = (author: WpAuthor) => {
	return {
		...author,
		link: {
			href: `/authors/${author.slug}`,
			text: `Read More`,
			title: `All Posts By ${author.name}`
		}
	};
};

function PostTitle(props: { title: string }) {
	return (
		<PostHeader
			level={1}
			tw="font-bold font-sans break-normal text-gray-900 pt-6 pb-2 text-3xl md:text-4xl"
			title={props.title}
		/>
	);
}

/**
 * Props for the "renderTitle" render prop.
 */
export type renderTitleProps = {
	title: string;
	post: WpPost;
};
type renderPropForTitle = (props: renderTitleProps) => JSX.Element;

export function PostTop(props: {
	post: WpPost;
	renderTitle?: (props: renderTitleProps) => JSX.Element;
	editMode?: boolean;
	onClickSettings?: () => void;
	renderBelowTitle?: () => JSX.Element;
	renderFeatured?: renderFeaturedImage;
}) {
	const { post, onClickSettings, renderFeatured } = props;
	const { title, published } = post;
	const featured: Image = post.featured
		? post.featured
		: {
				src: "http://placekitten.com/1280/640",
				alt: "A cat"
		  };
	return (
		<Fragment>
			<FeaturedImage featured={featured} renderFeatured={renderFeatured} />
			<div
				className="w-full px-4 md:px-6 text-xl text-gray-800 leading-normal"
				style={{ fontFamily: "Georgia,serif" }}
			>
				<div className="font-sans">
					<div className="text-base md:text-sm text-teal-500 font-bold inline-block">
						<span>
							{props.renderTitle ? (
								props.renderTitle({
									title: title.rendered,
									post
								})
							) : (
								<PostTitle title={title.rendered} />
							)}
							{props.renderBelowTitle ? (
								React.createElement(props.renderBelowTitle, {
									post
								})
							) : (
								<p className="text-sm md:text-base font-normal text-gray-600">
									Published {published}
								</p>
							)}
						</span>
					</div>
					{onClickSettings && (
						<div className={"inline-block"}>
							<button onClick={onClickSettings}>Settings</button>
						</div>
					)}
				</div>
			</div>
		</Fragment>
	);
}
export default function(props: {
	post: WpPost;
	renderTitle?: renderPropForTitle;
	renderContent?: renderPropForContent;
	onClickSettings?: () => void;
	renderBelowTitle?: () => JSX.Element;
	renderFeatured?: renderFeaturedImage;
}) {
	const {
		post,
		renderTitle,
		renderContent,
		onClickSettings,
		renderBelowTitle,
		renderFeatured
	} = props;
	const { tags, author, content } = post;
	return (
		<Article tw="container w-full md:max-w-3xl mx-auto pt-20">
			<hr className="border-b-2 border-gray-400 mb-8 mx-4" />
			<PostTop
				post={props.post}
				renderTitle={renderTitle}
				onClickSettings={onClickSettings}
				renderBelowTitle={renderBelowTitle}
				renderFeatured={renderFeatured}
			/>
			<PostContent content={content.rendered} renderContent={renderContent} />
			<Footer>
				{undefined !== tags && tags.length && (
					<div className="text-base md:text-sm text-gray-500 px-4 py-6">
						<PostTags tags={tags} />
					</div>
				)}
				<Author author={addAuthorLink(author)} />
			</Footer>
		</Article>
	);
}
