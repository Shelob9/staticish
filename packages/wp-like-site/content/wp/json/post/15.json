{"id":15,"date":"2019-12-26T18:09:36","date_gmt":"2019-12-26T18:09:36","guid":{"rendered":"http://localhost:3100/?p=15"},"modified":"2019-12-26T18:09:36","modified_gmt":"2019-12-26T18:09:36","slug":"other-author","status":"publish","type":"post","link":"http://localhost:3100/other-author/","title":{"rendered":"Other Author"},"content":{"rendered":"\n<h2> Are et dolore magna aliqua ?</h2>\n\n\n\n<p>Are et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.  </p>\n","protected":false},"excerpt":{"rendered":"","protected":false},"author":2,"featured_media":0,"comment_status":"open","ping_status":"open","sticky":false,"template":"","format":"standard","meta":[],"categories":[5],"tags":[6,7],"_links":{"self":[{"href":"http://localhost:3100/wp-json/wp/v2/posts/15"}],"collection":[{"href":"http://localhost:3100/wp-json/wp/v2/posts"}],"about":[{"href":"http://localhost:3100/wp-json/wp/v2/types/post"}],"author":[{"embeddable":true,"href":"http://localhost:3100/wp-json/wp/v2/users/2"}],"replies":[{"embeddable":true,"href":"http://localhost:3100/wp-json/wp/v2/comments?post=15"}],"version-history":[{"count":1,"href":"http://localhost:3100/wp-json/wp/v2/posts/15/revisions"}],"predecessor-version":[{"id":16,"href":"http://localhost:3100/wp-json/wp/v2/posts/15/revisions/16"}],"wp:attachment":[{"href":"http://localhost:3100/wp-json/wp/v2/media?parent=15"}],"wp:term":[{"taxonomy":"category","embeddable":true,"href":"http://localhost:3100/wp-json/wp/v2/categories?post=15"},{"taxonomy":"post_tag","embeddable":true,"href":"http://localhost:3100/wp-json/wp/v2/tags?post=15"}],"curies":[{"name":"wp","href":"https://api.w.org/{rel}","templated":true}]}}