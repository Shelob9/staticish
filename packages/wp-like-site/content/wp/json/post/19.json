{"id":19,"date":"2019-12-26T18:11:37","date_gmt":"2019-12-26T18:11:37","guid":{"rendered":"http://localhost:3100/?p=19"},"modified":"2019-12-26T18:11:47","modified_gmt":"2019-12-26T18:11:47","slug":"cats","status":"publish","type":"post","link":"http://localhost:3100/cats/","title":{"rendered":"Cats"},"content":{"rendered":"\n<figure class=\"wp-block-gallery aligncenter columns-2 is-cropped\"><ul class=\"blocks-gallery-grid\"><li class=\"blocks-gallery-item\"><figure><img src=\"http://localhost:3100/wp-content/uploads/2019/12/cat-statue.jpg\" alt=\"A statue of a cat\" data-id=\"10\" data-full-url=\"http://localhost:3100/wp-content/uploads/2019/12/cat-statue.jpg\" data-link=\"http://localhost:3100/post-about-cat/cat-statue/\" class=\"wp-image-10\" srcset=\"http://localhost:3100/wp-content/uploads/2019/12/cat-statue.jpg 800w, http://localhost:3100/wp-content/uploads/2019/12/cat-statue-300x300.jpg 300w, http://localhost:3100/wp-content/uploads/2019/12/cat-statue-150x150.jpg 150w, http://localhost:3100/wp-content/uploads/2019/12/cat-statue-768x768.jpg 768w\" sizes=\"(max-width: 800px) 100vw, 800px\" /><figcaption class=\"blocks-gallery-item__caption\">Statue Kitten</figcaption></figure></li><li class=\"blocks-gallery-item\"><figure><img src=\"http://localhost:3100/wp-content/uploads/2019/12/tiny-cat.jpg\" alt=\"A very tiny cat\" data-id=\"7\" data-full-url=\"http://localhost:3100/wp-content/uploads/2019/12/tiny-cat.jpg\" data-link=\"http://localhost:3100/post-with-cat-as-featured-image/tiny-cat/\" class=\"wp-image-7\"/><figcaption class=\"blocks-gallery-item__caption\">Kitten Kitten</figcaption></figure></li></ul></figure>\n\n\n\n<figure class=\"wp-block-image size-large\"><img src=\"http://localhost:3100/wp-content/uploads/2019/12/cat-statue.jpg\" alt=\"\" class=\"wp-image-10\" srcset=\"http://localhost:3100/wp-content/uploads/2019/12/cat-statue.jpg 800w, http://localhost:3100/wp-content/uploads/2019/12/cat-statue-300x300.jpg 300w, http://localhost:3100/wp-content/uploads/2019/12/cat-statue-150x150.jpg 150w, http://localhost:3100/wp-content/uploads/2019/12/cat-statue-768x768.jpg 768w\" sizes=\"(max-width: 800px) 100vw, 800px\" /><figcaption>Cat statue</figcaption></figure>\n\n\n\n<h2> Are et dolore magna aliqua ?</h2>\n\n\n\n<p>Are et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.  </p>\n","protected":false},"excerpt":{"rendered":"","protected":false},"author":1,"featured_media":0,"comment_status":"open","ping_status":"open","sticky":false,"template":"","format":"standard","meta":[],"categories":[3],"tags":[4],"_links":{"self":[{"href":"http://localhost:3100/wp-json/wp/v2/posts/19"}],"collection":[{"href":"http://localhost:3100/wp-json/wp/v2/posts"}],"about":[{"href":"http://localhost:3100/wp-json/wp/v2/types/post"}],"author":[{"embeddable":true,"href":"http://localhost:3100/wp-json/wp/v2/users/1"}],"replies":[{"embeddable":true,"href":"http://localhost:3100/wp-json/wp/v2/comments?post=19"}],"version-history":[{"count":1,"href":"http://localhost:3100/wp-json/wp/v2/posts/19/revisions"}],"predecessor-version":[{"id":20,"href":"http://localhost:3100/wp-json/wp/v2/posts/19/revisions/20"}],"wp:attachment":[{"href":"http://localhost:3100/wp-json/wp/v2/media?parent=19"}],"wp:term":[{"taxonomy":"category","embeddable":true,"href":"http://localhost:3100/wp-json/wp/v2/categories?post=19"},{"taxonomy":"post_tag","embeddable":true,"href":"http://localhost:3100/wp-json/wp/v2/tags?post=19"}],"curies":[{"name":"wp","href":"https://api.w.org/{rel}","templated":true}]}}