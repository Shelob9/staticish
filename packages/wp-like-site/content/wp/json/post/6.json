{"id":6,"date":"2019-12-26T18:07:27","date_gmt":"2019-12-26T18:07:27","guid":{"rendered":"http://localhost:3100/?p=6"},"modified":"2019-12-26T18:07:27","modified_gmt":"2019-12-26T18:07:27","slug":"post-with-cat-as-featured-image","status":"publish","type":"post","link":"http://localhost:3100/post-with-cat-as-featured-image/","title":{"rendered":"Post With Cat As Featured Image"},"content":{"rendered":"\n<blockquote class=\"wp-block-quote\"><p>Lab Rats Are Only Powerless Because They Don’t Understand That They’re In An Experiment</p><cite>&#8211; The OA</cite></blockquote>\n\n\n\n<p> Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labor.</p>\n\n\n\n<h2>Heading</h2>\n\n\n\n<p> re et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.  </p>\n\n\n\n<p> re et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.  </p>\n","protected":false},"excerpt":{"rendered":"<p>Lab Rats Are Only Powerless Because They Don’t Understand That They’re In An Experiment &#8211; The OA Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labor. Heading re et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. [&hellip;]</p>\n","protected":false},"author":1,"featured_media":7,"comment_status":"open","ping_status":"open","sticky":false,"template":"","format":"standard","meta":[],"categories":[3],"tags":[2],"_links":{"self":[{"href":"http://localhost:3100/wp-json/wp/v2/posts/6"}],"collection":[{"href":"http://localhost:3100/wp-json/wp/v2/posts"}],"about":[{"href":"http://localhost:3100/wp-json/wp/v2/types/post"}],"author":[{"embeddable":true,"href":"http://localhost:3100/wp-json/wp/v2/users/1"}],"replies":[{"embeddable":true,"href":"http://localhost:3100/wp-json/wp/v2/comments?post=6"}],"version-history":[{"count":1,"href":"http://localhost:3100/wp-json/wp/v2/posts/6/revisions"}],"predecessor-version":[{"id":8,"href":"http://localhost:3100/wp-json/wp/v2/posts/6/revisions/8"}],"wp:featuredmedia":[{"embeddable":true,"href":"http://localhost:3100/wp-json/wp/v2/media/7"}],"wp:attachment":[{"href":"http://localhost:3100/wp-json/wp/v2/media?parent=6"}],"wp:term":[{"taxonomy":"category","embeddable":true,"href":"http://localhost:3100/wp-json/wp/v2/categories?post=6"},{"taxonomy":"post_tag","embeddable":true,"href":"http://localhost:3100/wp-json/wp/v2/tags?post=6"}],"curies":[{"name":"wp","href":"https://api.w.org/{rel}","templated":true}]}}