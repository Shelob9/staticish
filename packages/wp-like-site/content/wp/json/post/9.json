{"id":9,"date":"2019-12-26T18:08:36","date_gmt":"2019-12-26T18:08:36","guid":{"rendered":"http://localhost:3100/?p=9"},"modified":"2019-12-26T18:08:36","modified_gmt":"2019-12-26T18:08:36","slug":"post-about-cat","status":"publish","type":"post","link":"http://localhost:3100/post-about-cat/","title":{"rendered":"Post ABout Cat"},"content":{"rendered":"\n<figure class=\"wp-block-image size-large\"><img src=\"http://localhost:3100/wp-content/uploads/2019/12/cat-statue.jpg\" alt=\"\" class=\"wp-image-10\" srcset=\"http://localhost:3100/wp-content/uploads/2019/12/cat-statue.jpg 800w, http://localhost:3100/wp-content/uploads/2019/12/cat-statue-300x300.jpg 300w, http://localhost:3100/wp-content/uploads/2019/12/cat-statue-150x150.jpg 150w, http://localhost:3100/wp-content/uploads/2019/12/cat-statue-768x768.jpg 768w\" sizes=\"(max-width: 800px) 100vw, 800px\" /><figcaption>A statue of a cat</figcaption></figure>\n\n\n\n<h2> Are et dolore magna aliqua ?</h2>\n\n\n\n<p>Are et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.  </p>\n","protected":false},"excerpt":{"rendered":"","protected":false},"author":1,"featured_media":0,"comment_status":"open","ping_status":"open","sticky":false,"template":"","format":"standard","meta":[],"categories":[3],"tags":[4],"_links":{"self":[{"href":"http://localhost:3100/wp-json/wp/v2/posts/9"}],"collection":[{"href":"http://localhost:3100/wp-json/wp/v2/posts"}],"about":[{"href":"http://localhost:3100/wp-json/wp/v2/types/post"}],"author":[{"embeddable":true,"href":"http://localhost:3100/wp-json/wp/v2/users/1"}],"replies":[{"embeddable":true,"href":"http://localhost:3100/wp-json/wp/v2/comments?post=9"}],"version-history":[{"count":1,"href":"http://localhost:3100/wp-json/wp/v2/posts/9/revisions"}],"predecessor-version":[{"id":12,"href":"http://localhost:3100/wp-json/wp/v2/posts/9/revisions/12"}],"wp:attachment":[{"href":"http://localhost:3100/wp-json/wp/v2/media?parent=9"}],"wp:term":[{"taxonomy":"category","embeddable":true,"href":"http://localhost:3100/wp-json/wp/v2/categories?post=9"},{"taxonomy":"post_tag","embeddable":true,"href":"http://localhost:3100/wp-json/wp/v2/tags?post=9"}],"curies":[{"name":"wp","href":"https://api.w.org/{rel}","templated":true}]}}