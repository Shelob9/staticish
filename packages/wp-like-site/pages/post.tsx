import React, { Fragment } from "react";
import Head from "next/head";
import { Post } from "@staticish/wp-api-to-static";
import { BlogPostPreview } from "@staticish/post-ui";
import "../css/index.css";

export default function() {
	const post = {
		id: 168826,
		date: "2019-12-17T11:16:33",
		date_gmt: "2019-12-17T16:16:33",
		guid: { rendered: "https://calderaforms.com/?p=168826" },
		modified: "2019-12-17T11:51:49",
		modified_gmt: "2019-12-17T16:51:49",
		slug: "2019-holiday-update",
		status: "publish",
		type: "post",
		link: "https://calderaforms.com/2019/12/2019-holiday-update/",
		title: { rendered: "A Quick Holiday Update!" },
		content: {
			rendered: "\n<h2>Bug fixes as a present are kind of like getting",
			protected: false
		},
		excerpt: {
			rendered:
				'<p>Bug fixes as a present are kind of like getting socks for Christmas&#8230; not the most thrilling thing, but when you need socks&#8230; The holidays are often a different kind of time for teams. We want to keep moving all the things forward, but it&#8217;s also healthy to slow down and appreciate family, too. The &hellip; <a href="https://calderaforms.com/2019/12/2019-holiday-update/" class="more-link">Continue reading <span class="screen-reader-text">A Quick Holiday Update!</span></a></p>\n',
			protected: false
		},
		author: 62578,
		featured_media: 168887,
		comment_status: "open",
		ping_status: "open",
		sticky: false,
		template: "",
		format: "standard",
		categories: [174],
		tags: [],
		meta: {}
	};

	return (
		<Fragment>
			<Head>
				<title>Next.js + Jest + TypeScript</title>
			</Head>
			<BlogPostPreview post={post} />
		</Fragment>
	);
}
