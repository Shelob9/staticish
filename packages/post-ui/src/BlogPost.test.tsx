import React from 'react';
import BlogPost from './BlogPost';
import { render } from '@testing-library/react';
import wpFactory from './factories/wpFactory';
import author from './factories/test-data/author';
import mediaItem from './factories/test-data/media';
import tag from './factories/test-data/tag';
import post from './factories/test-data/post';

describe('Blog post with factory', () => {
  const getters = {
    author: () => author,
    featured: () => mediaItem,
    tags: () => {
      //@ts-ignore
      return [tag];
    },
    published: () => post.date,
  };
  //@ts-ignore
  const factory = wpFactory(getters);
  const wpPost = factory.convertPost(post);
  test('It matches snapshot', () => {
    const { container } = render(<BlogPost post={wpPost} />);
    expect(container).toMatchSnapshot();
  });
});
