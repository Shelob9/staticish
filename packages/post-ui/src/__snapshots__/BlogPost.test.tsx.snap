// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blog post with factory It matches snapshot 1`] = `
<div>
  <article
    class="container w-full md:max-w-3xl mx-auto pt-20"
  >
    <hr
      class="border-b-2 border-gray-400 mb-8 mx-4"
    />
    <img
      alt="A cat"
      class="w-full"
      src="http://placekitten.com/1280/640"
    />
    <div
      class="w-full px-4 md:px-6 text-xl text-gray-800 leading-normal"
      style="font-family: Georgia,serif;"
    >
      <div
        class="font-sans"
      >
        <div
          class="text-base md:text-sm text-teal-500 font-bold inline-block"
        >
          <span>
            <header>
              <h1
                class="font-bold font-sans break-normal text-gray-900 pt-6 pb-2 text-3xl md:text-4xl"
              >
                A Quick Holiday Update!
              </h1>
            </header>
            <p
              class="text-sm md:text-base font-normal text-gray-600"
            >
              Published 
              2019-12-17T11:16:33
            </p>
          </span>
        </div>
      </div>
    </div>
    <div
      class="text-gray-700 text-base"
    >
      

      <h2>
        Bug fixes as a present are kind of like getting
      </h2>
    </div>
    <footer
      class="px-6 py-4"
    >
      <div
        class="text-base md:text-sm text-gray-500 px-4 py-6"
      >
        <div
          class=""
        >
          <span
            class="inline-block bg-black text-white rounded-full px-3 py-1 text-sm font-semibold text-gray-700"
          >
            #hiRoy
          </span>
        </div>
      </div>
      <div
        class="flex w-full items-center font-sans px-4 py-12"
      >
        <img
          alt="Avatar for Josh Pollock"
          class="w-10 h-10 rounded-full mr-4"
          src="https://secure.gravatar.com/avatar/90f0d6d84cc52508796d039eb81c4f06?s=96&d=mm&r=g"
        />
        <div
          class="flex-1 px-2"
        >
          <p
            class="text-base font-bold text-base md:text-xl leading-none mb-2"
          >
            Josh Pollock
          </p>
          <p
            class="text-gray-600 text-xs md:text-base"
          >
            Josh Pollock is VP of Engineering Experience at &lt;a href="https://saturdaydrive.io/"&gt;Saturday Drive&lt;/a&gt;. Previously, he was the founder of CalderaWP and the lead developer of Caldera Forms. He is also a WordPress core contributor and the author of two books about WordPress development.
          </p>
          <p
            class="text-gray-600 text-xs md:text-base"
          >
            <a
              class="text-teal-500 no-underline hover:underline"
              href="https://calderaforms.com/author/josh/"
            >
              nodejs
            </a>
          </p>
        </div>
        <div
          class="justify-end"
        >
          <button
            class="bg-transparent border border-gray-500 hover:border-teal-500 text-xs text-gray-500 hover:text-teal-500 font-bold py-2 px-4 rounded-full"
          >
            Read More
          </button>
        </div>
      </div>
    </footer>
  </article>
</div>
`;
